### PR TITLE
[libc++][modules] Removes module testing.

### PR DESCRIPTION
**Pull request by mordante**
This removes the entire modules testing infrastructure.

The current infrastructure uses CMake to generate the std and std.compat module. This requires quite a bit of plumbing and uses CMake. Since CMake introduced module support in CMake 3.26, modules have a higher CMake requirement than the rest of the LLVM project. (The LLVM project requires 3.20.) The main motivation for this approach was how libc++ generated its modules. Every header had its own module partition. This was changed to improve performance and now only two modules remain. The code to build these can be manually crafted.

A followup patch will reenable testing modules, using a different approach.

### CODE REVIEW

**Review by ldionne — COMMENTED** [review]
Obviously I think that much simplification is nice, but shouldn't we wait until we have the replacement before we merge this removal?

**Comment by mordante** [comment]
> Obviously I think that much simplification is nice, but shouldn't we wait until we have the replacement before we merge this removal?

I have a replacement and I want to stack that on top of this patch once the CI is green. The intention is to disable module testing for a few minutes. However I want to avoid mixing the patches to give a better view of which change does what.

**Review by ldionne — APPROVED** [review]
I see, SGTM. Thanks!

**Pull request merged**
